bulldozer/merge.go

```diff
@@ -57,6 +57,39 @@ func NewGitHubMerger(client *github.Client) Merger {
 }
 
 func (m *GitHubMerger) Merge(ctx context.Context, pullCtx pull.Context, method MergeMethod, msg CommitMessage) (string, error) {
+	if method == FastForwardOnly {
+		return m.ffOnlyMerge(ctx, pullCtx)
+	}
+
+	return m.defaultMerge(ctx, pullCtx, method, msg)
+}
+
+func (m *GitHubMerger) ffOnlyMerge(ctx context.Context, pullCtx pull.Context) (string, error) {
+	// ff-only merge is accomplished by calling Git.UpdateRef with the force parameter set to false, and the new commit hash for the base branch's pointer
+
+	base, _ := pullCtx.Branches()
+
+	ref, _, err := m.client.Git.GetRef(ctx, pullCtx.Owner(), pullCtx.Repo(), fmt.Sprintf("refs/heads/%s", base))
+	if err != nil {
+		return "", err
+	}
+
+	headCommitSHA := pullCtx.HeadSHA()
+	ref.Object.SHA = &headCommitSHA
+
+	newRef, _, err := m.client.Git.UpdateRef(ctx, pullCtx.Owner(), pullCtx.Repo(), ref, false)
+	if err != nil {
+		return "", err
+	}
+
+	if newRef.GetObject().GetSHA() != headCommitSHA {
+		return "", fmt.Errorf("Expected reference to be updated to SHA %s, but instead it points to %s", headCommitSHA, newRef.GetObject().GetSHA())
```

Comment:
Is this the response when a fast-forward is not possible, i.e. GitHub returns "success", but the ref doesn't actually change? If so, it would be helpful to indicate that in this message, maybe as `could not fast-foward $oldsha to $newsha`.

Reply:
honestly, I'm not sure if this situation will ever happen.  If it would return a different hash here, the call to UpdateRef _should_ return an error, but since it gives us the new reference as a return value, figured I'd check it just in case.